### ISSUE TITLE
win_iis_webapppool - not able to set all the properties

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#25467]
_From @AseemKumar on August 16, 2016 11:31_

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
win_iis_webapppool

##### ANSIBLE VERSION
```
ansible 2.1.1.0
```

##### CONFIGURATION

Default config
##### OS / ENVIRONMENT

ansible is running on Ubuntu Trusty and the systems being managed are linux and Windows.
##### SUMMARY

Tried to change the value of Identity under Process Model in advanced settings of an appPool and the change did not take effect. Tried to change the value of other properties under Process Model and none gets changed, even though the output of ansible says changed. IIS being managed is 7.5 version. Can only few properties be modified of IIS from the ansible module as of now? If yes, please specify that in the documentation.
##### STEPS TO REPRODUCE

```
  - name: Set Identity Type to Network Service.
    win_iis_webapppool:
      name: UIAPI
      attributes: identityType:2
      state: restarted
```

##### EXPECTED RESULTS

The Identity Type of the app pool should have changed to Network Service, but it did not.
##### ACTUAL RESULTS

The value of identityType did not change, even though ansible reported that it has been changed.

```
<ServerName> ESTABLISH WINRM CONNECTION FOR USER: UserName on PORT 5985 TO ServerName
<ServerName> EXEC Set-StrictMode -Version Latest
(New-Item -Type Directory -Path $env:temp -Name "ansible-tmp-1471345999.53-80318070680631").FullName | Write-Host -Separator '';
<ServerName> PUT "/tmp/tmpm8UpFI" TO "C:\Users\userID\AppData\Local\Temp\ansible-tmp-1471345999.53-80318070680631\win_iis_webapppool.ps1"
<ServerName> EXEC Set-StrictMode -Version Latest
Try
{
& 'C:\Users\userID\AppData\Local\Temp\ansible-tmp-1471345999.53-80318070680631\win_iis_webapppool.ps1'
}
Catch
{
$_obj = @{ failed = $true }
If ($_.Exception.GetType)
{
$_obj.Add('msg', $_.Exception.Message)
}
Else
{
$_obj.Add('msg', $_.ToString())
}
If ($_.InvocationInfo.PositionMessage)
{
$_obj.Add('exception', $_.InvocationInfo.PositionMessage)
}
ElseIf ($_.ScriptStackTrace)
{
$_obj.Add('exception', $_.ScriptStackTrace)
}
Try
{
$_obj.Add('error_record', ($_ | ConvertTo-Json | ConvertFrom-Json))
}
Catch
{
}
Echo $_obj | ConvertTo-Json -Compress -Depth 99
Exit 1
}
Finally { Remove-Item "C:\Users\userID\AppData\Local\Temp\ansible-tmp-1471345999.53-80318070680631" -Force -Recurse -ErrorAction SilentlyContinue }
changed: [ServerName] => {"attributes": {"identityType": "2"}, "changed": true, "info": {"attributes": {"CLRConfigFile": "", "applicationPoolSid": "S-1-5-82-
3266571588-702921383-1992572147-801121604-1441276117", "autoStart": true, "enable32BitAppOnWin64": false, "enableConfigurationOverride": true, "managedPipelineMode": 0, "
managedRuntimeLoader": "webengine4.dll", "managedRuntimeVersion": "v4.0", "name": "UIAPI", "passAnonymousToken": true, "queueLength": 1000, "startMode": 1, "state": 1},
 "name": "UIAPI", "state": "Started"}, "invocation": {"module_name": "win_iis_webapppool"}}

PLAY RECAP *********************************************************************
ServerName    : ok=1    changed=1    unreachable=0    failed=0
```

_Copied from original issue: ansible/ansible-modules-extras#2732_

**Comment by dagwieers** [comment]
_From @ansibot on August 16, 2016 11:41_



@whenrik ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on September 11, 2016 14:53_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on October 6, 2016 17:50_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on October 22, 2016 10:8_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on November 6, 2016 10:44_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on November 22, 2016 10:51_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on December 6, 2016 23:7_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
cc @jborean93 @jhawkesworth
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jhawkesworth** [comment]
This is a feature request really.  Would be useful to have though.  If I recall the app pool and the identity associated with it are configured in slightly different parts of the configuration for IIS.

Will edit summary so this is a feature request.

The attributes you can modify are returned in the 'info' section of the return results (example below).

```
ansible iis--m win_iis_webapppool -a 'name=DefaultAppPool' -v
Using /home/jon/ansible/ansible.cfg as config file
Thursday 08 June 2017  10:27:51 +0100 (0:00:00.416)       0:00:00.416 *********
iis001 | SUCCESS => {
    "attributes": {},
    "changed": false,
    "failed": false,
    "info": {
        "attributes": {
            "CLRConfigFile": "",
            "applicationPoolSid": "S-1-5-82-3006700770-424185619-1745488364-794895919-4004696415",
            "autoStart": true,
            "enable32BitAppOnWin64": true,
            "enableConfigurationOverride": true,
            "managedPipelineMode": "Integrated",
            "managedRuntimeLoader": "webengine4.dll",
            "managedRuntimeVersion": "v2.0",
            "name": "DefaultAppPool",
            "passAnonymousToken": true,
            "queueLength": 1000,
            "startMode": "OnDemand",
            "state": "Started"
        },
        "name": "DefaultAppPool",
        "state": "Started"
    }
}


```

**Comment by pfar1** [comment]
I think this is a primarly a bug in addition to a feature request.  I have been trying to get this to work myself and had some problems.

The PS code using the Set-ItemProperty is taking a key, and there is a list of documented options with the module as outlined above, however.... you can use dot notation to pass in other keys that are not documented.

The documented params are the top-level application pool settings, anything that is under a child is not documented.

If you wanted to set the Load User Profile parameter, you would need to pass the full path with its parent, which is processModel.loadUserProfile: False or the identity would be processModel.identityType:2, so whilst those additional options are not documented, you can set them.  I dont think there is anything further required to expose them, i think using processModel.loadUserProfile is acceptable vs just loadUserProfile...

Note: They are not returned in the output because only the top level params are being returned.

Example of setting child params.
```
- name: Create X Application pool
  win_iis_webapppool:
   name: X
   state: started
attributes: 'name:ABE|processModel.loadUserProfile:False|processModel.identityType:ApplicationPoolIdentity'
```
Perhaps the full paths should be documented to make it easier to understand as it initially appears by intent or not, that only certain parameters can be set.

Now, for the bug......
It doesnt seem to want to set any value with an Int type, including the documented queuelength.

I have tried various params including queuelength, processModel.maxProcesses etc.

I have taken the PS code and executed locally on a server and i get the same result as executing it via Ansible when i am trying to set the queuelength or other params with ints.  They return the following error via Ansible:

```
fatal: [mymachine]: FAILED! => {
    "attributes": {
        "name": "X",
        "processModel.identityType": "ApplicationPoolIdentity",
        "processModel.loadUserProfile": "False",
        "processModel.maxProcesses": "5"
    },
    "changed": false,
    "failed": true,
    "msg": "Specified cast is not valid."
}
```
I get the same cast error via the powershell script output on the server.

Finally, if i execute the command in a PS window without the script:

`Set-ItemProperty ("IIS:\AppPools\X") queueLength 1201`

This does actually work perfectly fine, it is exactly the same as:
`Set-ItemProperty ("IIS:\AppPools\" + $name) $newParameter.Key $newParameter.Value`

I have tried casting it in the script etc, someone with better skills needs to look at this and due to what is happening i think this is actually a bug.  Some better documentation is needed to bring to light the fact you can set child parameters via dot notation.

Thanks
Paul

**Comment by pfar1** [comment]
Also, just to add...

The module doesnt appear to work properly (although no error) when broke over multiple lines like so...

- name: Create ABE Application pool
  win_iis_webapppool:
   name: X
   state: started
   attributes: |
     'managedRuntimeVersion:{{iis_apppool_runtimever}}|
      enable32BitAppOnWin64:{{iis_apppool_enable32bit}}|
      name:X|
      queueLength:{{iis_apppool_queuelen}}|
      startMode:{{iis_apppool_startmode}}|
      cpu.limit:{{iis_apppool_cpu_limit_pct}}|
      cpu.action:{{iis_apppool_cpu_limit_action}}|
      cpu.resetInterval:{{iis_apppool_cpu_limit_interval}}|
      cpu.smpAffinitized:{{iis_apppool_cpu_procaff_enabled}}|
      cpu.smpProcessorAffinityMask:{{iis_apppool_cpu_procaff_mask}}'

**Comment by pfar1** [comment]
Just to add... i have now tested setting the various attributes and have found the following are the ones the module fails to set:
```
   queueLength:
   processModel.pingResponseTime:
   processModel.pingInterval:
   processModel.shutdownTimeLimit:
   processModel.startupTimeLimit:
   failure.rapidFailProtectionInterval:
   failure.rapidFailProtectionMaxCrashes:
   recycling.periodicRestart.memory:
   recycling.periodicRestart.privateMemory:
   recycling.periodicRestart.time:
   recycling.periodicRestart.requests:
```

These are either int64's or timespans and the normal error is "The specified cast is not valid".

I also believe there is an issue with the pre-check:
`if(-not $currentParamVal -or ($currentParamVal -as [String]) -ne $newParameter.Value)`

currentParamVal never contains just the value, it is normally at currentParamVal.Value and hence it always fails and attempts to set the new value.

Thanks
Paul

**Comment by jborean93** [comment]
@pfar1 thanks for the info, I'll add some tests to prove my upcoming changes work with it. Have already come across the pre-check issue with both your comment and it handling `$currentParamVal` being set to `0`.

**Comment by jborean93** [comment]
@pfar1 I've got a PR raised to fix this issue https://github.com/ansible/ansible/pull/25869. I've been able to set those fields and others, you can have a look at the tests and new examples to see how to set timespan values. It would be appreciative if you could test the PR and verify it.

**Comment by pfar1** [comment]
@jborean93 i would love to get involved and test this for you, unfortunately adding the comments was my first time using github so i am not entirely sure on what i need to do? if you can point me in the right direction, more than happy to help.

I am using Ansible 2.3.0.0 on Ubuntu, presumably i need to add these changes to my installation somehow?

Thanks
Paul

**Comment by jborean93** [comment]
@pfar1, no worries we all start somewhere. You have 2 options available to use for testing

* Create a file called `win_iis_webapppool.ps1` in a folder and run `export ANSIBLE_LIBRARY=/path/to/folder` before running your playbook, or
* Modify the existing module file under your Ansible install. This would be `/path/to/ansible/lib/ansible/modules/windows/win_iis_webapppool.ps1`.

The file you create or modify should have the contents of https://github.com/jborean93/ansible/blob/2f209ddc784c1b6ae37510ff4e6d1b5696e8d1c7/lib/ansible/modules/windows/win_iis_webapppool.ps1.

**Comment by pfar1** [comment]
@jborean93 I have renamed the original and dropped the new ps1 file in and test.

I am getting an error:

`
TASK [ms-iis-apppool : Assuring TEST Application pool] ******************************************************************************
task path: /etc/ansible/roles/ms-iis-apppool/tasks/main.yml:60
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/windows/win_iis_webapppool.ps1
<192.168.11.198> ESTABLISH WINRM CONNECTION FOR USER: user@domain.local on PORT 5986 TO x.x.x.x
EXEC (via pipeline wrapper)
 [WARNING]: Failure using method (v2_runner_on_failed) in callback plugin (<ansible.plugins.callback.default.CallbackModule object
at 0x7ff55a672e10>): deprecated() argument after ** must be a mapping, not list

        to retry, use: --limit @/etc/ansible/plays/roletest.retry
`
I have also modified my task as follows:
```
- name: "Assuring {{iis_apppool_name}} Application pool"
  win_iis_webapppool:
   name: "{{iis_apppool_name}}"
   state: started
   attributes:
     name:{{iis_apppool_name}}
     managedRuntimeVersion:{{iis_apppool_managedRuntimeVersion}}
     enable32BitAppOnWin64:{{iis_apppool_enable32BitAppOnWin64}}
     managedPipelineMode:{{iis_apppool_managedPipelineMode}}
     startMode:{{iis_apppool_startMode}}
     queuelength:{{iis_apppool_queuelength}}
```
Thanks
Paul

**Comment by pfar1** [comment]
FYI, whilst it is a warning, it does create the application pool but it isnt setting in values.

**Comment by jborean93** [comment]
Hmm you might be coming across issues with the deprecation tasks that are run in the callback, must have been some changes since 2.3 that is causing this. You can still test your task by running

```
git clone https://github.com/jborean93/ansible.git
cd ansible
git checkout win_iis_webapppool-tests
source hacking/env-setup
ansible-playbook my-playbooks.yml
```

This will clone the branch with the changes and the new Ansible version for your testing.

A side note, you can set one variable for attributes and use that in your task like below:

```
# vars.yml
iis_attributes:
  name: Pool
  managedRuntimeVersion: v4.0
  enable32BitAppOnWin64: True
  managedPipelineMode: Integrated
  startMode: AutoStart
  queuelength: 1

# playbook.yml
- name: create app pool {{iis_apppool_name}}
  win_iis_webapppool:
    name: "{{iis_apppool_name}}"
    state: started
    attributes: "{{iis_attributes}}"
```

**Comment by pfar1** [comment]
@jborean93 ok running the clone now and thanks for the vars info...

Fyi my version is 2.3.0.0..

**Comment by pfar1** [comment]
@jborean93 almost.....

No longer getting the warning, and it is setting the values i have tried below, except for queuelength again.

It isnt returning any error at all this time, but it hasnt set the queuelength.



```
- name: "Assuring {{iis_apppool_name}} Application pool"
  win_iis_webapppool:
   name: "{{iis_apppool_name}}"
   state: started
   attributes:
     name: "{{iis_apppool_name}}"
     managedRuntimeVersion: "{{iis_apppool_managedRuntimeVersion}}"
     enable32BitAppOnWin64: "{{iis_apppool_enable32BitAppOnWin64}}"
     managedPipelineMode: "{{iis_apppool_managedPipelineMode}}"
     startMode: "{{iis_apppool_startMode}}"
     queuelength: 2000
```

**Comment by jborean93** [comment]
I'll give it a test tomorrow, I wouldn't have thought it would be case sensitive (some posts online indicate it might be) but can you try queueLength instead as that's how it is set here https://www.iis.net/configreference/system.applicationhost/applicationpools/add#005. If that's the case I'll update he docs for the PR to reference that fact.

**Comment by pfar1** [comment]
@jborean93 good spot... its case sensitive... queueLength now works.

Ill test the others and report back.. If all is ok, what do i do? just let you know here or add comments etc to the PR?

**Comment by pfar1** [comment]
@jborean93 i have now completed the testing..

The following all now set ok.

```
   queueLength:
   processModel.pingResponseTime:
   processModel.pingInterval:
   processModel.startupTimeLimit:
   failure.rapidFailProtectionInterval:
   failure.rapidFailProtectionMaxCrashes:
   recycling.periodicRestart.memory:
   recycling.periodicRestart.privateMemory:
   recycling.periodicRestart.time:
   recycling.periodicRestart.requests:
```

I am unable to set processModel.shutdownTimeLimit, this is the only value that doesnt seem to want to set now.

Also, just to add, if something fails such as the above, or the incorrect cased queueLength, Ansible is reporting it as changed and not reporting any issues.

I have also spotted another issue with the return, it isnt reporting the correct state of the Application Pool.  My Windows Process Activation service was stopped and this caused the state of all AppPools to be stopped.  Even when the AppPool and service is stopped it still sets everything ok but when it returns it says the state of the AppPool is started, but it isnt, it is stopped.


```
changed: [ansy02.dmz.local] => {
    "attributes": {
        "cpu.action": "NoAction",
        "cpu.limit": 0,
        "cpu.numaNodeAffinityMode": "Soft",
        "cpu.numaNodeAssignment": "MostAvailableMemory",
        "cpu.processorGroup": 0,
        "cpu.smpAffinitized": false,
        "cpu.smpProcessorAffinityMask": 4294967295,
        "cpu.smpProcessorAffinityMask2": 4294967295,
        "enable32BitAppOnWin64": false,
        "failure.autoShutdownExe": null,
        "failure.autoShutdownParams": null,
        "failure.loadBalancerCapabilities": "HttpLevel",
        "failure.orphanActionExe": null,
        "failure.orphanActionParams": null,
        "failure.orphanWorkerProcess": false,
        "failure.rapidFailProtection": false,
        "failure.rapidFailProtectionInterval": "00:00:01:00",
        "failure.rapidFailProtectionMaxCrashes": 20,
        "managedPipelineMode": "Integrated",
        "managedRuntimeVersion": "v4.0",
        "name": "test",
        "processModel.identityType": "ApplicationPoolIdentity",
        "processModel.idleTimeout": 0,
        "processModel.idleTimeoutAction": "Terminate",
        "processModel.loadUserProfile": false,
        "processModel.maxProcesses": 2,
        "processModel.pingInterval": "00:00:10",
        "processModel.pingResponseTime": "00:00:10",
        "processModel.pingingEnabled": true,
        "processModel.shutdownTimeLimit": "00:02:30}",
        "processModel.startupTimeLimit": "00:00:10",
        "queueLength": 1200,
        "recycling.disallowOverlappingRotation": false,
        "recycling.disallowRotationOnConfigChange": false,
        "recycling.periodicRestart.memory": 1024,
        "recycling.periodicRestart.privateMemory": 1024,
        "recycling.periodicRestart.requests": 100,
        "recycling.periodicRestart.time": "00:01:00",
        "startMode": "AlwaysRunning"
    },
    "changed": true,
    "failed": false,
    "info": {
        "attributes": {
            "CLRConfigFile": "",
            "applicationPoolSid": "S-1-5-82-4046446318-2148771789-4022891813-4260434434-115948061",
            "autoStart": true,
            "enable32BitAppOnWin64": false,
            "enableConfigurationOverride": true,
            "managedPipelineMode": "Integrated",
            "managedRuntimeLoader": "webengine4.dll",
            "managedRuntimeVersion": "v4.0",
            "name": "ABE",
            "passAnonymousToken": true,
            "queueLength": 1200,
            "startMode": "AlwaysRunning",
            "state": null
        },
        "cpu": {
            "action": "NoAction",
            "limit": 0,
            "numaNodeAffinityMode": "Soft",
            "numaNodeAssignment": "MostAvailableMemory",
            "processorGroup": 0,
            "resetInterval": {
                "Days": 0,
                "Hours": 0,
                "Milliseconds": 0,
                "Minutes": 5,
                "Seconds": 0,
                "Ticks": 3000000000,
                "TotalDays": 0.003472222222222222,
                "TotalHours": 0.08333333333333333,
                "TotalMilliseconds": 300000,
                "TotalMinutes": 5,
                "TotalSeconds": 300
            },
            "smpAffinitized": false,
            "smpProcessorAffinityMask": 4294967295,
            "smpProcessorAffinityMask2": 4294967295
        },
        "failure": {
            "autoShutdownExe": "",
            "autoShutdownParams": "",
            "loadBalancerCapabilities": "HttpLevel",
            "orphanActionExe": "",
            "orphanActionParams": "",
            "orphanWorkerProcess": false,
            "rapidFailProtection": false,
            "rapidFailProtectionInterval": {
                "Days": 0,
                "Hours": 0,
                "Milliseconds": 0,
                "Minutes": 1,
                "Seconds": 0,
                "Ticks": 600000000,
                "TotalDays": 0.0006944444444444444,
                "TotalHours": 0.016666666666666666,
                "TotalMilliseconds": 60000,
                "TotalMinutes": 1,
                "TotalSeconds": 60
            },
            "rapidFailProtectionMaxCrashes": 20
        },
        "name": "test",
        "processModel": {
            "identityType": "ApplicationPoolIdentity",
            "idleTimeout": {
                "Days": 0,
                "Hours": 0,
                "Milliseconds": 0,
                "Minutes": 0,
                "Seconds": 0,
                "Ticks": 0,
                "TotalDays": 0,
                "TotalHours": 0,
                "TotalMilliseconds": 0,
                "TotalMinutes": 0,
                "TotalSeconds": 0
            },
            "idleTimeoutAction": "Terminate",
            "loadUserProfile": false,
            "logEventOnProcessModel": "IdleTimeout",
            "logonType": "LogonBatch",
            "manualGroupMembership": false,
            "maxProcesses": 2,
            "password": "",
            "pingInterval": {
                "Days": 0,
                "Hours": 0,
                "Milliseconds": 0,
                "Minutes": 0,
                "Seconds": 10,
                "Ticks": 100000000,
                "TotalDays": 0.00011574074074074073,
                "TotalHours": 0.0027777777777777775,
                "TotalMilliseconds": 10000,
                "TotalMinutes": 0.16666666666666666,
                "TotalSeconds": 10
            },
            "pingResponseTime": {
                "Days": 0,
                "Hours": 0,
                "Milliseconds": 0,
                "Minutes": 0,
                "Seconds": 10,
                "Ticks": 100000000,
                "TotalDays": 0.00011574074074074073,
                "TotalHours": 0.0027777777777777775,
                "TotalMilliseconds": 10000,
                "TotalMinutes": 0.16666666666666666,
                "TotalSeconds": 10
            },
            "pingingEnabled": true,
            "setProfileEnvironment": true,
            "shutdownTimeLimit": {
                "Days": 0,
                "Hours": 0,
                "Milliseconds": 0,
                "Minutes": 1,
                "Seconds": 30,
                "Ticks": 900000000,
                "TotalDays": 0.0010416666666666667,
                "TotalHours": 0.024999999999999998,
                "TotalMilliseconds": 90000,
                "TotalMinutes": 1.5,
                "TotalSeconds": 90
            },
            "startupTimeLimit": {
                "Days": 0,
                "Hours": 0,
                "Milliseconds": 0,
                "Minutes": 0,
                "Seconds": 10,
                "Ticks": 100000000,
                "TotalDays": 0.00011574074074074073,
                "TotalHours": 0.0027777777777777775,
                "TotalMilliseconds": 10000,
                "TotalMinutes": 0.16666666666666666,
                "TotalSeconds": 10
            },
            "userName": ""
        },
        "recycling": {
            "disallowOverlappingRotation": false,
            "disallowRotationOnConfigChange": false,
            "logEventOnRecycle": "Time,Requests,Schedule,Memory,IsapiUnhealthy,OnDemand,ConfigChange,PrivateMemory",
            "periodicRestart": {
                "memory": 1024,
                "privateMemory": 1024,
                "requests": 100,
                "time": {
                    "Days": 0,
                    "Hours": 0,
                    "Milliseconds": 0,
                    "Minutes": 1,
                    "Seconds": 0,
                    "Ticks": 600000000,
                    "TotalDays": 0.0006944444444444444,
                    "TotalHours": 0.016666666666666666,
                    "TotalMilliseconds": 60000,
                    "TotalMinutes": 1,
                    "TotalSeconds": 60
                }
            }
        },
        "state": "started"
    }
}
```

**Comment by jborean93** [comment]
Hey thanks for the feedback, I believe your shutdown time isn't working as your value is not a correct timespan i.e. `"00:02:30}"` when it should just be `"00:02:30"`. For the state issue look under `info.attributes.state` and you can see it is null and not `started`. The parameter `info.state` is just a reflection of what was passed into the module. I'll have a look at why the state is returning as null in this case.

I suppose I should do a check first hand to verify the attributes that we set exist and if they don't fail out as well as verifying what we set is actually set. Unfortunately the different IIS versions seem to have different levels of strictness which makes it hard but I'll give it a shot.

**Comment by pfar1** [comment]
Hi @jborean93, ok apologies for not spotting the mistake with the timestamp... had an extra } on the end of the variable.  It works as expected now.

How long does it typically take to get this released into the main Ansible product? is there a way i can consume this in the interim, bearing in mind the depreciation issue i had?

Thanks
Paul

**Comment by jborean93** [comment]
@pfar1 it all depends on how long it takes for it to get merged in and 2.4 to be out, usually it is on a ~ 4 month release cycle. You can use the powershell file and add it to your library but you may need to modify it and remove all the warning cmdlets that are found in there to get it to work.

If you get some time are you able to post your brief findings on the PR and either some suggestions or thoughts so others can see. You may also be interested, I am rewritting the win_iis_* modules, keep a lookout for those PRs if you are using any of them.

**Comment by pfar1** [comment]
@jborean93 yes sure ill add some comments to it shortly.

I guess we may be in touch again then, i am new to Ansible and am trying to get a web server to deploy from VMware and into Production with code, purely from Ansible.  I have cracked the VM deployment and IIS install and now AppPool configuration..  Next up is creating websites etc so i might have more to add.

Right now, other than the incorrect state, this particular module does everything, i dont think there is anything missing.

**Comment by pfar1** [comment]
@jborean93 what do you mean by warning cmdlets? is it the -whatif ? or the catch statements? dont suppose you could supply a useable version could you ;)

**Comment by jborean93** [comment]
@pfar1, you would have to remove `Add-DeprecationWarning`, I believe there is only one at the beginning of the script for this PR. That should hopefully be enough to get it working on an older version of Ansible.

**Comment by pfar1** [comment]
@jborean93 that looks to have done the trick, thanks!

**Comment by jborean93** [comment]
Closing issue as PR has been merged that fixes these issues.
